### PR TITLE
[WIP] Output stack graph to PR bodies

### PR DESCRIPTION
**Pull request by oleg-codaio**
When creating/updating/syncing stacks, also include the latest stack of changes in each pull request description. This essentially replicates the `av stack tree` visualization in PRs:

image

This is off by default and can be enabled via `pullRequest.writeStack` (for those not yet using a merge queue).

### Work in progress

- [X] Update `av stack sync` to include the current stack in all open PR descriptions
- [X] Update `av pr create` to include the current stack
- [x] Fix bug where `av pr create` doesn't include itself until re-sync
- [x] Fix bug where `av stack submit` doesn't include the stack
- [ ] Figure out how to keep merged PRs included in descendant stacks (until the whole stack tree is merged into trunk)
- [x] Enable setting branch prefix via config

### CODE REVIEW

**Comment by oleg-codaio** [comment]
Oops - not ready for review yet. Goal is to bring this CLI on par with https://github.com/ejoffe/spr, https://github.com/facebook/sapling, https://github.com/VirtusLab/git-machete, etc. if that's something the maintainers are interested in.

**Comment by jainankit** [comment]
@oleg-codaio thanks for putting this together 🙏  I think it'll be easier if you break this down into smaller changes for review. After all we are working on stacked PRs :)

Let me know if you need any help.

**Comment by oleg-codaio** [comment]
Sounds good! I can send out smaller PRs once I've had a chance to address some issues and dogfood the feature more.

However, do stacked PRs work properly across forks? I don't think there's a way to submit (at least not now) PRs using `av` to a different repo, and there might also be a GitHub limitation that requires write access to properly chain them.

**Comment by jainankit** [comment]
You are right, it won't work well with forked repo. Feel free to continue on this change or separate out if you find a logical way to segment. We can take a look when you are ready.

**Comment by oleg-codaio** [comment]
I've made some more progress and fixed some other minor bugs on the way. As far as getting this reviewed, I can start sending out smaller target PRs soon.

If you have any early feedback on the stack visuals, open to it here as well. But here's how I approached it thus far:

For the majority of stacks (based on my experience with stacking and reviewing other devs' stacked PRs), they'll start at the trunk branch and create a single, linear stack of PRs (i.e., a degenerate tree). I think for those it's nice to see the bottommost branch at the top with the trunk all the way at the bottom, like so:

image

This is also similar to the `av stack tree` command shows the tree.

Once a stack gets more complex with different fork points, it becomes harder to represent the stack tree this way (unless you show sibling branches to the side somehow and prune the tree). The current implementation instead prints the output as a left-to-right tree:

image

Unlike what `av stack tree` does, this visualization isn't "sorted" to have the current branch be along the top. The reason for this is to have a consistent tree across the whole stack of PRs, so you can navigate from one branch to another more predictably.

If you have any feedback on this, let me know.

**Comment by draftcode** [comment]
Yep. It's hard to replicate what current `av stack tree` (or `git log --graph`) does with HTML lists. Using the second option (top-to-bottom tree) looks fine to me.

**Comment by oleg-codaio** [comment]
Sounds good! One other thing - instead of having the stack on the bottom of the PR, we could also have it on the top in a collapsable `<detail>` header. Any preference on one or the other, or providing an option for the user to pick their preference?

image

image

**Comment by draftcode** [comment]
It's fine in any cases. Please pick one. If there's a FR for choosing another way, we can add a configuration for the style.

File: cmd/av/commit_amend.go [hunk 1]
```diff
@@ -78,7 +78,7 @@ var commitAmendCmd = &cobra.Command{
 		// Even if it's not configured, there's no need to fetch/push
 		state.Config.NoFetch = true
 		state.Config.NoPush = true
-		err = actions.SyncStack(ctx, repo, client, tx, branchesToSync, state)
+		err = actions.SyncStack(ctx, repo, client, tx, branchesToSync, state, actions.WithLocalOnly())
```
Comment (oleg-codaio) [inline cmd/av/commit_amend.go:81]: Split this out into https://github.com/aviator-co/av/pull/264.

Comment (oleg-codaio) [inline cmd/av/commit_amend.go:81]: Split this out into https://github.com/aviator-co/av/pull/266.

**Comment by oleg-codaio** [comment]
Closing; most of the changes have been merged in and this is the final one: https://github.com/aviator-co/av/pull/271

**Pull request closed without merging**